### PR TITLE
Test analytics

### PR DESCRIPTION
**Pull request by kristiandupont**

### CODE REVIEW

**Comment by kristiandupont** [comment]
@fealebenpae and/or @nirinchev take a look at this if you have time. I think we should either merge it in or get rid of it. (There are merge issues atm, but I will fix those if we deem this valuable)

File: RealmWeaver/Analytics.cs [hunk 1]
```diff
@@ -53,75 +52,82 @@ namespace RealmWeaver
...
+        string AppId { get; }
+        string OsName { get; }
+        string OsVersion { get; }
+        string TargetOs { get; }
+        string RealmVersion { get; }
+    }
 
+    public class AnalyticsData : IAnalyticsData
+    {
+        public string MixPanelToken => "ce0fac19508f6c8f20066d345d360fd0";
```
Comment (nirinchev) [inline RealmWeaver/Analytics.cs]: This should go below `private readonly ModuleDefinition _moduleDefinition` :)

File: RealmWeaver/Analytics.cs [hunk 2]
```diff
@@ -53,75 +52,82 @@ namespace RealmWeaver
...
+                    var searcher = new ManagementObjectSearcher(scope, query);
+
+                    var uuid = searcher.Get().Cast<ManagementObject>().FirstOrDefault()?["UUID"] as string;
+                    return string.IsNullOrEmpty(uuid) ? null : Encoding.UTF8.GetBytes(uuid);
                 }
-                catch
+                else  // Assume OS X if not Windows.
                 {
-                    return "UNKNOWN";
+                    var macs = from nic in NetworkInterface.GetAllNetworkInterfaces()
```
Comment (nirinchev) [inline RealmWeaver/Analytics.cs]: Fairly minor, but this would look prettier with extension methods

File: RealmWeaver/Analytics.cs [hunk 3]
```diff
@@ -186,48 +166,67 @@ private static string SHA256Hash(byte[] bytes)
...
-            }
+            return SHA256Hash(Encoding.UTF8.GetBytes(@string));
+        }
 
-            internal static bool ComputeSwVersionOSNameAndVersion(out string name, out string version)
-            {
-                version = null;
-                return RunSwVersion("-productName", out name) && RunSwVersion("-productVersion", out version);
-            }
+        private static string JsonTemplate => @"{
```
Comment (fealebenpae) [inline RealmWeaver/Analytics.cs]: Please make it a field again.

**Pull request closed without merging**